applications/nrf_desktop/doc/battery_charger.rst

```diff
@@ -0,0 +1,53 @@
+.. _battery_charger:
+
+Battery charger module
+######################
+
+The ``battery_charger`` module is responsible for battery charging control.
+
+Module Events
+*************
+
++------------------+---------------------+---------------------+-------------------------+------------------+
+| Source Module    | Input Event         | This Module         | Output Event            | Sink Module      |
++==================+=====================+=====================+=========================+==================+
+|                  |                     | ``battery_charger`` | ``battery_state_event`` | :ref:`led_state` |
++------------------+---------------------+                     +-------------------------+------------------+
+| :ref:`usb_state` | ``usb_state_event`` |                     |                         |                  |
++------------------+---------------------+---------------------+-------------------------+------------------+
+
+Configuration
+*************
+
+The module implemented in ``battery_charger.c`` uses the Zephyr :ref:`zephyr:gpio` driver to control and monitor battery charging. For this reason, you should set :option:`CONFIG_GPIO` option.
+
+Set option ``CONFIG_DESKTOP_BATTERY_CHARGER_DISCRETE`` to enable the module (by default ``CONFIG_DESKTOP_BATTERY_CHARGER_NONE`` is selected - module is disabled).
+
+Following module configuration options are available:
+* ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_PIN`` - pin to which CSO (charging status) output from charger is connected
+* CSO pin pull:
+  * ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_PULL_NONE`` - disabled (default option)
+  * ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_PULL_UP`` - up
+  * ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_PULL_DOWN`` - down
+* ``CONFIG_DESKTOP_BATTERY_CHARGER_ENABLE_PIN`` - pin which enables battery charging
+* ``CONFIG_DESKTOP_BATTERY_CHARGER_ENABLE_INVERSED`` - inverse charging enable signal
+* ``CONFIG_DESKTOP_BATTERY_CHARGER_CSO_FREQ`` - CSO pin state switching fequency when charging error occurrs [Hz]
+
+Implementation details
+**********************
+
+The module disables battery charging when USB is suspended (required by USB specification) or disconnected.
```

Comment:
What does it mean `(required by USB specification)`? That suspending USB is required by the USB spec?
```suggestion
As required by the USB specification, the module disables the battery charging when the USB is suspended. In addition, it also disables charging when the USB is disconnected.
```

Reply:
I meant that charging has to be disabled when USB is in suspended state (device has to reduce power consumption from USB).